packages/commcare/src/Adaptor.js

```diff
@@ -203,29 +162,35 @@ export function submit(formData) {
  * @returns {Operation}
  */
 export function fetchReportData(reportId, params, postUrl) {
-  return http.get(`api/v0.5/configurablereportdata/${reportId}/`, {
-    query: function (state) {
-      console.log(
-        'getting from url: '.concat(
-          state.configuration.baseUrl,
-          `/api/v0.5/configurablereportdata/${reportId}/`
-        )
-      );
-      console.log('with params: '.concat(params));
-      return params;
-    },
-    callback: function (state) {
-      var reportData = state.response.body;
-      return post(postUrl, {
-        body: reportData,
-      })(state).then(function (state) {
-        delete state.response;
+  return state => {
+    const path = `/a/${state.configuration.applicationName}/api/v0.5/configurablereportdata/${reportId}/`;
+
+    console.log('with params: '.concat(JSON.stringify(params)));
+
+    return request({
```

Comment:
I've just had an idea!

The nested promises here have always seemed weird to me. Nothing wrong with it, just wierd and a little hard to follow the logic.

Does it get any clearer if we re-write this whole chain with async/await?

Reply:
This is untested bit I was thinking about this:
```
const { body: reportData } = await request({
  state,
  method: 'GET',
  path,
  authType: 'basic',
})
const result = await request({
  state,
  method: 'POST',
  path: postUrl,
  params,
  data: reportData,
  authType: 'basic',
});

delete result.response;
console.log('fetchReportData succeeded.');
console.log('Posted to: '.concat(postUrl));
return prepareNextState(state, {});
```